src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -205,25 +205,31 @@ private void SetCollectionObjectWithCollectionMember(ref object collection, Coll
                     collection = ReflectionCreateObject(collectionType);
                 }
 
-                if (typeof(IList).IsAssignableFrom(collectionType))
+                AddObjectsIntoTargetCollection(collection, collectionMember, collectionType);
+            }
+        }
+
+        private static void AddObjectsIntoTargetCollection(object targetCollection, List<object> sourceCollection, Type targetCollectionType)
+        {
+            var targetList = targetCollection as IList;
+            if (targetList != null)
+            {
+                foreach (var item in sourceCollection)
                 {
-                    foreach (var item in collectionMember)
-                    {
-                        ((IList)collection).Add(item);
-                    }
+                    targetList.Add(item);
                 }
-                else
+            }
+            else
+            {
+                foreach (var item in sourceCollection)
                 {
-                    foreach (var item in collectionMember)
+                    MethodInfo addMethod = targetCollectionType.GetMethod("Add");
```

Comment:
Is it really necessary to repeat the `Add` method lookup via reflection on every iteration? Can it be resolved once instead and then cached?